src/core/src/model.cpp

```diff
@@ -238,7 +247,33 @@ void ov::Model::validate_nodes_and_infer_types() const {
         } else if (const auto& read_value = std::dynamic_pointer_cast<ngraph::op::ReadValueBase>(node)) {
             pair_checker[read_value->get_variable().get()].cnt_read_val++;
         }
+
+        // Detect names collisions only for nodes with autogenerated names
+        const auto friendly_name = node->get_friendly_name();
+        if (unique_friendly_names.find(friendly_name) == unique_friendly_names.end()) {
+            unique_friendly_names.insert(friendly_name);
+            if (node->m_friendly_name.empty())
+                nodes_with_possible_conflicts[friendly_name] = node.get();
+        } else if (node->m_friendly_name.empty()) {
+            // We have a conflict with autogenerated name
+            nodes_with_conflicts.emplace_back(node.get());
+        } else if (nodes_with_possible_conflicts.find(friendly_name) != nodes_with_possible_conflicts.end()) {
+            // We have a conflict with autogenerated name
+            nodes_with_conflicts.emplace_back(nodes_with_possible_conflicts[friendly_name]);
+        }
     }
+
+    // Resolve names collisions
+    for (const auto& node : nodes_with_conflicts) {
+        size_t idx = 2;
+        const auto friendly_name = node->get_friendly_name();
+        while (unique_friendly_names.find(friendly_name + "_" + std::to_string(idx)) != unique_friendly_names.end())
+            idx++;
+        const auto new_friendly_name = friendly_name + "_" + std::to_string(idx);
+        node->set_friendly_name(new_friendly_name);
```

Comment:
After we set the node name it automatically becomes not autogenerated, so next time when we try to resolve conflicts we won't change it.

Reply:
Discussed offline